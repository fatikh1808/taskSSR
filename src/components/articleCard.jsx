import React from "react"
import { Card } from "antd"
import "antd/dist/antd.css"
import { Typography } from "antd"

const { Meta } = Card
const { Text } = Typography

export const ArticleCard = ({ id, author_name, title, rubric, image, author_last_name, backgroundColor, width, borderRadius }) => {

  return (
    <Card
      hoverable
      style={{ borderRadius, backgroundColor, width}}
      cover={<img alt="example" src={image}/>}
    >
      <Meta title={title}/>
      <Text style={{ marginRight: 5 }}>{author_name}</Text>
      <Text>{author_last_name}</Text>
      <Meta description={rubric}/>
      <Meta description={<a href={`/${rubric}/${id}`}>More</a>}
      />
    </Card>
  )
}

ArticleCard.defaultProps = {
  backgroundColor: null,
  width: "80%",
  borderRadius: 5
};
