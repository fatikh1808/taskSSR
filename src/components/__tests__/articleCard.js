import React from "react"
import renderer from "react-test-renderer"

import {ArticleCard} from "../articleCard"

describe("ArticleCard", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<ArticleCard siteTitle="Default Starter" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})