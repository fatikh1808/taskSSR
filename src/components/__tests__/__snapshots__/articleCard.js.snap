// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard renders correctly 1`] = `
<div
  className="ant-card ant-card-bordered ant-card-hoverable"
  style={
    Object {
      "borderRadius": 5,
      "width": "80%",
    }
  }
>
  <div
    className="ant-card-cover"
  >
    <img
      alt="example"
    />
  </div>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ant-card-meta"
    />
    <span
      className="ant-typography"
      style={
        Object {
          "WebkitLineClamp": null,
          "marginRight": 5,
        }
      }
    >
      
    </span>
    <span
      className="ant-typography"
      style={
        Object {
          "WebkitLineClamp": null,
        }
      }
    >
      
    </span>
    <div
      className="ant-card-meta"
    />
    <div
      className="ant-card-meta"
    >
      <div
        className="ant-card-meta-detail"
      >
        <div
          className="ant-card-meta-description"
        >
          <a
            href="/undefined/undefined"
          >
            More
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
